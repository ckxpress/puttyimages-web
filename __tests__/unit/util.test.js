const { isNumber, getPageInfo } = require('../../server/util');

test('isNumber', () => {
  expect(isNumber(1)).toBeTruthy();
  expect(isNumber(0.1)).toBeTruthy();
  expect(isNumber('1')).toBeTruthy();
  expect(isNumber('0.1')).toBeTruthy();

  expect(isNumber('.1')).toBeFalsy();
  expect(isNumber('..1')).toBeFalsy();
});

test('getPageInfo', () => {
  const baseUrl = 'http://localhost:3000/api/search?q=keyword';
  const after = { offset: 10, q: 'keyword' };
  const before = { offset: 0, q: 'keyword' };
  const totalCount = 10;
  expect(() => getPageInfo()).toThrow();
  expect(() => getPageInfo({ baseUrl })).toThrow();
  expect(() => getPageInfo({ baseUrl, totalCount })).not.toThrow();

  expect(getPageInfo({ after, baseUrl, before, totalCount })).toMatchSnapshot();
  expect(getPageInfo({ after, baseUrl, totalCount })).toMatchSnapshot();
  expect(getPageInfo({ baseUrl, before, totalCount })).toMatchSnapshot();
  expect(getPageInfo({ baseUrl, totalCount })).toMatchSnapshot();
});
