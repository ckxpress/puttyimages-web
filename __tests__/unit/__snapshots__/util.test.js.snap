// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPageInfo 1`] = `
Object {
  "endCursor": "eyJvZmZzZXQiOjEwLCJxIjoia2V5d29yZCJ9",
  "hasNextPage": true,
  "hasPrevPage": true,
  "next": "http://localhost:3000/api/search?q=keyword&after=eyJvZmZzZXQiOjEwLCJxIjoia2V5d29yZCJ9",
  "prev": "http://localhost:3000/api/search?q=keyword&before=eyJvZmZzZXQiOjAsInEiOiJrZXl3b3JkIn0",
  "startCursor": "eyJvZmZzZXQiOjAsInEiOiJrZXl3b3JkIn0",
  "totalCount": 10,
}
`;

exports[`getPageInfo 2`] = `
Object {
  "endCursor": "eyJvZmZzZXQiOjEwLCJxIjoia2V5d29yZCJ9",
  "hasNextPage": true,
  "hasPrevPage": false,
  "next": "http://localhost:3000/api/search?q=keyword&after=eyJvZmZzZXQiOjEwLCJxIjoia2V5d29yZCJ9",
  "prev": null,
  "startCursor": null,
  "totalCount": 10,
}
`;

exports[`getPageInfo 3`] = `
Object {
  "endCursor": null,
  "hasNextPage": false,
  "hasPrevPage": true,
  "next": null,
  "prev": "http://localhost:3000/api/search?q=keyword&before=eyJvZmZzZXQiOjAsInEiOiJrZXl3b3JkIn0",
  "startCursor": "eyJvZmZzZXQiOjAsInEiOiJrZXl3b3JkIn0",
  "totalCount": 10,
}
`;

exports[`getPageInfo 4`] = `
Object {
  "endCursor": null,
  "hasNextPage": false,
  "hasPrevPage": false,
  "next": null,
  "prev": null,
  "startCursor": null,
  "totalCount": 10,
}
`;
